src/main/java/uk/gov/hmcts/dts/mytime/models/LeaveType.java

```diff
@@ -0,0 +1,22 @@
+package uk.gov.hmcts.dts.mytime.models;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import uk.gov.hmcts.dts.mytime.exceptions.EnumConversionException;
+
+public enum LeaveType {
+    FLEXI,
+    HOLIDAY;
+
+    @JsonCreator
+    public static LeaveType create(String value) {
+        if (value == null) {
+            throw new IllegalArgumentException();
```

Comment:
Can we put a message in here? Also when would this be null?

Reply:
This @JsonCreator is used on the endpoint request when create a new leave request. The Leave type is passed through the HTTP as a string but we are representing it as enum in the repository and the code. Hence we have this manual enum conversion to throw the error when validating the request body. As it is a string in request, it can be null if Leave type is not passed in from the request body. Have added a message in the exception :)